rust/phoenix-channel/src/lib.rs

```diff
@@ -393,6 +393,32 @@ where
                                 reason: "unknown error (bad event?)".to_owned(),
                             }))
                         }
+                        Payload::ControlMessage(ControlMessage::PhxClose(_)) => {
```

Comment:
@thomaseizinger could you double check that this is fine?

Reply:
Is there a reason you had to duplicate this code and could not tie into the state machine further up? If you call the following with some variant of `Error`, it will automatically reconnect.

```rust
self.reconnect_on_transient_error(...);
```

Reply:
just missed this, my bad